src/newton_ac_powerflow.jl

```diff
@@ -0,0 +1,731 @@
+            x,
+            data,
+            get(kwargs, :maxIter, DEFAULT_NR_MAX_ITER),
+        )
+        @info("PowerFlow solve converged, the results have been stored in the system")
+    else
+        @error("The powerflow solver returned convergence = $(converged)")
+    end
+
+    #Restore original per unit base
+    PSY.set_units_base_system!(system, settings_unit_cache)
+
+    return converged
+end
+
+"""
+Similar to solve_powerflow!(sys) but does not update the system struct with results.
+Returns the results in a dictionary of dataframes.
+
+## Examples
+
+```julia
+res = solve_powerflow(sys)
+# Passing NLsolve arguments
+res = solve_powerflow(sys, method=:newton)
+```
+"""
+function solve_powerflow(
+    pf::ACPowerFlow{<:ACPowerFlowSolverType},
+    system::PSY.System;
+    kwargs...,
+)
+    #Save per-unit flag
+    settings_unit_cache = deepcopy(system.units_settings.unit_system)
+    #Work in System per unit
+    PSY.set_units_base_system!(system, "SYSTEM_BASE")
+
+    data = PowerFlowData(
+        pf,
+        system;
+        check_connectivity = get(kwargs, :check_connectivity, true),
+    )
+
+    converged, V, Sbus_result = _ac_powereflow(data, pf; kwargs...)
+    x = _calc_x(data, V, Sbus_result)
+
+    if converged
+        @info("PowerFlow solve converged, the results are exported in DataFrames")
+        df_results = write_results(pf, system, data, x)
+    else
+        df_results = missing
+        @error("The powerflow solver returned convergence = $(converged)")
+    end
+
+    #Restore original per unit base
+    PSY.set_units_base_system!(system, settings_unit_cache)
+
+    return df_results
+end
+
+# Multiperiod power flow - work in progress
+function solve_powerflow(
+    pf::ACPowerFlow{<:ACPowerFlowSolverType},
+    data::ACPowerFlowData,
+    system::PSY.System;
+    kwargs...,
+)
+    #Save per-unit flag
+    settings_unit_cache = deepcopy(system.units_settings.unit_system)
+    #Work in System per unit
+    PSY.set_units_base_system!(system, "SYSTEM_BASE")
+
+    sorted_time_steps = sort(collect(keys(data.timestep_map)))
+    # preallocate results
+    ts_converged = zeros(Bool, 1, length(sorted_time_steps))
+    ts_V = zeros(Complex{Float64}, length(data.bus_type[:, 1]), length(sorted_time_steps))
+    ts_S = zeros(Complex{Float64}, length(data.bus_type[:, 1]), length(sorted_time_steps))
+
+    results = Dict()
+
+    for t in sorted_time_steps
+        converged, V, Sbus_result =
+            _ac_powereflow(data, pf; time_step = t, kwargs...)
+        ts_converged[1, t] = converged
+        ts_V[:, t] .= V
+        ts_S[:, t] .= Sbus_result
+
+        # temporary implementation that will need to be improved:
+        if converged
+            x = _calc_x(data, V, Sbus_result)
+            results[data.timestep_map[t]] = write_results(pf, system, data, x)
+        else
+            results[data.timestep_map[t]] = missing
+        end
+
+        #todo: implement write_results for multiperiod power flow
```

Comment:
@rbolgaryn is this still TODO?

Reply:
Resolution: @rbolgaryn deletes this whole function, it is no longer needed

Reply:
deleted the function